File: ncd/src/main/java/org/intelehealth/ncd/search/pager/CategoryPagerAdapter.java
package org.intelehealth.ncd.search.pager;

import android.app.Activity;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.List;

public class CategoryPagerAdapter extends FragmentStateAdapter {

    private List<Fragment> fragmentList;

    public CategoryPagerAdapter(FragmentActivity activity, List<Fragment> fragmentList) {
        super(activity);
        this.fragmentList = fragmentList;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getItemCount() {
        return fragmentList.size();
    }
}

File: ncd/src/main/java/org/intelehealth/ncd/category/pager/CategoryPagerAdapter.kt
package org.intelehealth.ncd.category.pager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class CategoryPagerAdapter(
    fragmentActivity: FragmentActivity,
    private val fragments: List<Fragment>
) : FragmentStateAdapter(fragmentActivity) {

    private val fragmentMap = mutableMapOf<Int, Fragment>()

    override fun getItemCount(): Int = fragments.size

    override fun createFragment(position: Int): Fragment {
        val fragment = fragments[position]
        fragmentMap[position] = fragment
        return fragment
    }

    fun getFragment(position: Int): Fragment? {
        return fragmentMap[position]
    }
}